src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  static alive = [];
+  health = 100;
 
-class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (this instanceof Carnivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health <= 0 && Animal.alive.indexOf(animal) >= 0) {
+        Animal.alive.splice(Animal.alive.indexOf(animal), 1);
+      }
+    }
+  }
```

Comment:
The `bite` method is defined in the `Animal` class, but according to the task description, it should only be present in the `Carnivore` class. Move this method to the `Carnivore` class and ensure it only affects `Herbivore` instances and checks if the `Herbivore` is not hiding.